Realm/RLMRealm.h

```diff
@@ -495,6 +496,22 @@ __deprecated_msg("Use `performMigrationForConfiguration:error:`") NS_REFINED_FOR
  */
 + (BOOL)performMigrationForConfiguration:(RLMRealmConfiguration *)configuration error:(NSError **)error;
 
+/**
+ Creates a `RLMThreadHandover` from the passed in objects that can be safely imported on a separate thread.
+
+ Used to pass objects between threads in a thread-safe manner.
+
+ @warning Note that the returned `ThreadHandover` must be imported at most once, and that the current version of the
+ Realm will remain pinned until this package is imported or deallocated.
+
+ @param objects The objects to pass between threads.
+ @return A package that can be imported on a separate thread.
+
+ @see RLMThreadHandover
+ */
+- (RLMThreadHandover *)exportThreadHandoverWithObjects:(NSArray<id<RLMThreadConfined>> *)objects
```

Comment:
Completely ignoring naming altogether, we have quite a few choices for how we want to structure this API.

## Export
### Package Initializer
```swift
let package = ThreadSafePackage(exporting: [foo, bar], from: realm)
```
```swift
let package = ThreadSafePackage(items: [foo, bar]) // infer Realm from items
```

### Realm Method
```swift
let package = realm.exportThreadSafePackage(containing: [foo, bar])
```
```swift
let package = realm.export(items: [foo, bar])
```
```swift
let package = realm.package(items: [foo, bar])
```

## Import
### Throwing Method Returning Named Realm-Items Tuple
```swift
let (realm, items) = try! package.importOnCurrentThread() // or `import()` or `unpack()`
try! realm.write {
    for x in items {
        realm.delete(x)
    }
}
```

### Auto-Importing Convenience Properties
```swift
// power user feature for handling errors
// otherwise, implicitly binds to current thread on first access, crashing on error
try! package.bindToCurrentThread()

try! package.realm.write {
    for x in package.items {
        package.realm.delete(x)
    }
}
```

### Realm Initializer
```swift
// implicitly binds package to current thread on init, allowing handling of failure
let realm = try! Realm(from: package)
try! realm.write {
    for x in package.items {
        realm.delete(x)
    }
}
```